src/docs/arch-disagg-layered.dox

```diff
@@ -0,0 +1,107 @@
+/*! @arch_page arch-disagg-layered Layered Tables
+
+Layered tables are a new access method in WiredTiger designed to support distributed systems that
+leverage shared storage for the majority of a table's data. This mechanism enables a single node to
+perform writes to the table, while other nodes maintain a consistent view by tracking recent changes.
+Having nodes follow along with recent changes allows for two key features:
+- Follower nodes can serve read operations on recent data without needing to access the shared
+storage directly.
+- Follower nodes can quickly transition to take over write responsibilities if the current leader
+(writer) becomes unavailable.
+
+@section layered-arch High-level architecture
+
+These semantics are achieved by using two underlying constituents: a shared stable table and an
+ingest table. The node responsible for writing to shared storage operates directly on a single
+shared table and has the guarantee that no other node will perform writes to that table.
+All other nodes will also have an ingest table, which holds a short-term view of recent changes.
+
+This view is layered over the shared stable table, so any operations on a node with both an ingest
+and stable table will first look for a result in the ingest table, and if no result is found, fall
+back on the shared table.
+
+Layered tables are primarily for Disaggregated Storage Clusters (DSC) but can also be used outside
+of DSC for testing purposes, in particular, via local Page and Log Mock (PALM) infrastructure.
+
+\image html shared_cluster.png
+
+Stable Table:
+- A table shared across nodes in the cluster
+- Backed by a block manager that uses shared storage and supports having multiple different
+WiredTiger nodes share the same underlying data structure
+
+Ingest Table:
+- A short-term in-memory table that stores data that has been changed more recently than is
+available in the stable table
+- Supersedes checkpointed content in the stable table
+- This table is specific to a particular node
+
+@section layered-api-restrictions API restrictions
+
+Layered tables support all the regular tables operations, with exceptions listed here:
+
+- All writes, from all replicas, must be timestamped
+- All nodes must use the same URI for the same table
+- All nodes must use byte-wise identical keys, values, and timestamps for the same records.
+- Prepared transactions will behave as described in the design for distributed transactions in
+disaggregated storage
+- Durability is provided by the logical log service, not WiredTiger
+    - Commit durability (using write-ahead log) is not supported for Layered Tables.
+    - Checkpoint durability is asynchronous. The completion of a `WT_SESSION::checkpoint()` call
+    does not guarantee that the resulting checkpoint is on storage and visible to other nodes yet.
```

Comment:
I'm a bit iffy on this one. This is my understanding:

Completion of `WT_SESSION::checkpoint()` guarantees the checkpoint is persisted to shared storage and recorded in shared metadata. Other nodes can only see it after they pick up this checkpoint from shared storage

Note: A checkpoint is considered complete once dirty data in the stable table has been reconciled, the stable table is flushed to the page/log service, and the corresponding metadata entry is persisted to the shared metadata file.

Pls correct me if I've misinterpreted something.